Assets/BossRoom/Scripts/Shared/Net/NetworkVariableLobbyState.cs

```diff
@@ -0,0 +1,230 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using Unity.Netcode;
+using UnityEngine;
+
+namespace BossRoom
+{
+    [Serializable]
+    public class NetworkVariableLobbyState : NetworkVariableBase, IEnumerable, IEnumerator
```

Comment:
why not use a state like NetworkTransform is using? this would simplify your serialization code. You could still have a separate lobby state class.

Reply:
@VALERE91Unity

Reply:
I think it was because of the absolute need to be unmanaged. And once something is managed in a struct (event an array or a string) the whole struct becomes managed and cannot fit in a NetVariable anymore.